torch/distributed/nn/functional.py

```diff
@@ -94,20 +116,23 @@ def all_gather(tensor, group=dist.group.WORLD):
     return _AllGather.apply(group, tensor)
 
 
-def all_to_all(tensors, group=dist.group.WORLD):
+def all_to_all(output_tensor_list, input_tensor_list, group=dist.group.WORLD):
```

Comment:
I'm wondering if we should support `all_to_all_single` as well here? The reason is that I believe this `all_to_all` API might probably lead to some additional copies in our sharded linear implementation.

Reply:
Sure. I think this shouldn't be hard. I can do that.

Reply:
@mrshenli Was wondering if you had some thoughts around these BC breaking API changes? It makes these APIs more inline with the ones in distributed_c10d.py and I was wondering if this is fine since my understanding is that we haven't really officially released these APIs? (despite having them on our docs page).

Reply:
Yep, they are not yet officially release, and it should be fine to introduce BC-breaking changes. I created a PR to remove those APIs from our docs. 

Also cc @cbalioglu

Reply:
oops, forgot to add a pointer to the PR: https://github.com/pytorch/pytorch/pull/69011